azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java

```diff
@@ -232,11 +233,16 @@ private void setupMultiExecutorMode() {
    */
   @Override
   public void setupExecutors() throws ExecutorManagerException {
+    checkMultiExecutorMode();
     this.activeExecutors.setupExecutors();
   }
 
-  private boolean isMultiExecutorMode() {
-    return isMultiExecutorMode(this.azkProps);
+  private void checkMultiExecutorMode() {
```

Comment:
Have you considered retiring this config? What are the downsides?

Reply:
+1  retiring this config.

Reply:
I wanted to make sure that users don't accidentally start azkaban with a new version thinking that their configured `executor.host` would be used. This will make it clear for the users that local mode won't work any more and they have to update their conf to start using the only supported mode (where the executor hostnames are automatically resolved on executor side & read from the db table on azkaban-web).

Eventually this code can be removed once confident enough that all active users have already updated to some version new enough to have this change, so that they must've acknowledgingly switched to multi-executor mode.

Reply:
LGTM.

Could you add a TODO to retire this required config?

Reply:
> Could you add a TODO to retire this required config?

@HappyRay, done.